admin/auth_token.go

```diff
@@ -175,20 +175,27 @@ func (t *magicAuthToken) OwnerID() string {
 
 // IssueMagicAuthTokenOptions provides options for IssueMagicAuthToken.
 type IssueMagicAuthTokenOptions struct {
-	ProjectID       string
-	TTL             *time.Duration
-	CreatedByUserID *string
-	Attributes      map[string]any
-	FilterJSON      string
-	Fields          []string
-	State           string
-	DisplayName     string
-	Internal        bool
-	Resources       []database.ResourceName
+	ProjectID                string
+	TTL                      *time.Duration
+	CreatedByUserID          *string
+	Attributes               map[string]any
+	FilterJSON               string
+	Fields                   []string
+	State                    string
+	DisplayName              string
+	Internal                 bool
+	Resources                []database.ResourceName
+	TransitiveAccessResource *database.ResourceName // Optional, used for transitive access
 }
 
 // IssueMagicAuthToken generates and persists a new magic auth token for a project.
 func (s *Service) IssueMagicAuthToken(ctx context.Context, opts *IssueMagicAuthTokenOptions) (AuthToken, error) {
+	if opts.TransitiveAccessResource != nil {
+		if len(opts.Resources) > 0 || opts.FilterJSON != "" || len(opts.Fields) > 0 {
```

Comment:
I think we still need filter JSON support for explores/canvases (although not reports since the filter is available in the report spec)? E.g. if you create a public URL for a canvas with filters on it, then we need to capture the filters and push them down to the transitive access to the underlying metrics views.

Reply:
see above [comment](https://github.com/rilldata/rill/pull/7815#discussion_r2298609478)